src/utils/Dom.ts

```diff
@@ -733,6 +733,22 @@ export class Dom {
     return $$(<HTMLElement>this.el.cloneNode(deep));
   }
 
+  /**
+   * Determine if an element support a particular native DOM event.
+   * @param eventName The event to evaluate. Eg: touchstart, touchend, click, scroll.
+   */
+  public supportsEvent(eventName: string): boolean {
+    const eventToEvaluate = `on${eventName}`;
+    let isSupported = eventToEvaluate in this.el;
+
+    if (!isSupported && this.el.setAttribute) {
+      this.el.setAttribute(eventToEvaluate, 'return;');
```

Comment:
Now that I see this line, I am rethinking the above comment, and how it plays with the next line.

Does setting the attribute value to 'return` create a function?

Also, looking at the test, for the case of `foo`, we enter this block, but why do we emerge as `false`?

Reply:
It attempts to create an empty function that does nothing, and then check if it actually got hooked as a function on the element (it checks if typeof eventName == function), or simply as a string (DOM API is fun like that ;) )

Reply:
Ok, I see, then a follow-up would be why is trying to add a function important? If we emit an unsupported/unhandled event name, then nothing should happen.

Or is there code that is accessing the function via eventname and trying to trigger it (`el[eventName]()`)?

If this is the case, but we still end up with a string in certain browsers, then have we solved the problem?

Reply:
It's not trying to "add" anything to use to handler down the line.

Because some browser (IE) won't behave properly, we need to "hack" stuff a bit and use a proxy method to determine if the element supports the even or not. I guess that tells me this should be added as a comment ;)